### ISSUE TITLE
[java] UnusedPrivateField on member annotated with lombok @Delegate

### ISSUE THREAD

**Issue by RiotR1cket** [pmd/pmd#1703]
**Affects PMD Version:**  6.12.0

**Rule:** UnusedPrivateField

**Description:**
Lombok `@Delegate` on a member, means that lombok will generate methods into the class which mirror the method signatures of that member variable, and the bodies of the methods are just a simple delegation to the member. Obviously it makes use of the member variable; so it should not trigger PMD's `UnusedPrivateField` rule, but it does.

https://projectlombok.org/features/Delegate.html

**Code Sample demonstrating the issue:**

```
public class Foo {
    @Delegate private Bar bar;
}
```

**Running PMD through:** *Maven* -- but with dependency override to PMD version 6.12.0.

P.S. I think there could be a workaround, using `violationSuppressXPath` to suppress the `UnusedPrivateField` on members with the `Delegate` annotation. But I haven't yet figured out the XPath to match it. If an XPath wizard wants to share the right incantation it would be much appreciated!

**Comment by RiotR1cket** [comment]
It seems like maybe a good fix would be to update the [AbstractLombokAwareRule](https://github.com/pmd/pmd/blob/master/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/AbstractLombokAwareRule.java) class to be made aware of the `lombok.experimental` annotations.

**Comment by jsotuyod** [comment]
@RiotR1cket thanks! You are absolutely right, we did not look into lombok experimental annotations when building that; and yes, adding them in `AbstractLombokAwareRule` should suffice.

If you feel up to it, can you submit a PR?
